### PR TITLE
Add utm_params column to users table

### PR DESCRIPTION
**Pull request by artursmirnov**
Required for this ticket:https://travisci.assembla.com/spaces/Web/tickets/283

### CODE REVIEW

File: db/main/structure.sql [hunk 1]
```diff
@@ -2878,7 +2878,8 @@ CREATE TABLE public.users (
     redacted_at timestamp without time zone,
     preferences jsonb DEFAULT '{}'::jsonb,
     vcs_type character varying DEFAULT 'GithubUser'::character varying,
-    vcs_id character varying
+    vcs_id character varying,
+    utm_params character varying
```
Comment (pavel-d) [inline db/main/structure.sql]: are you going to store all UTM params in a single column as a string?

Comment (artursmirnov) [reply db/main/structure.sql]: Yes, this was the format that we've agreed on with @shairyar while discussing the sync with SF. Although, I'm totally open for opinions and discussions here.

**Comment by artursmirnov** [comment]
@pavel-d thank you for review and feedback! This data is only needed for synchronization with SF database. We don't have any plans on using it internally. So the data format has been chosen mainly to support the sync. Generally, I agree that separate table with data in JSON format would have been more optimal, useful and flexible in terms of potential future internal use. But I'm not sure how it would work for the sync (e.g. if it would require additional conversion, then it won't probably worth it). @shairyar what do you think?

**Comment by pavel-d** [comment]
@artursmirnov, if I understand correctly, UTM is actually a hash (utm_source, utm_medium, utm_campaign, utm_term and utm_content are possible keys). I'm just not sure that it's a good idea to store it in a string.

**Comment by artursmirnov** [comment]
@pavel-d you're right, essentially it's a hash, but it initially comes to `travis-web` as a string (e.g. `utm_source=source&utm_campaign=campaign-name&utm_medium=medium` and so forth). The intention was to put it to the DB as is, so that it syncs to SF and then gets extracted there to the format that SF needs. Does it make sense to you?

**Comment by pavel-d** [comment]
@artursmirnov, we will need to parse the query string anyway, so, I don't see any reason not to parse it from the very beginning. Regarding a separate table, I would prefer this option for several reasons: it will keep the users table (which is quite big) clean of non-essential data, if we ever decide to move/delete marketing data we can simply export/drop the entire table, in addition, `utm_params` will not be queried every time user model is loaded. But, it's just my preference, I will leave the decision up to you and Shairyar :)

**Comment by artursmirnov** [comment]
@pavel-d thank you for the feedback, it totally make sense to me! I really like your suggestion and would be happy to implement it, unless @shairyar has any objections from SF sync point of view.

In your opinion, in case if we decide to go with the separate table for UTM data, would it be better to store different parameters in their own columns or keep them all in one column as JSON data? The first approach would allow us to eventually use this data via joins, add indexes and so on (I really doubt we'll ever need this tho). On the other hand, the second approach is more flexible as it won't require DB scheme changes to add more UTM fields. WDYT?

**Comment by pavel-d** [comment]
@artursmirnov, I don't think we will ever need joins or indexes, I think one column is fine

**Comment by artursmirnov** [comment]
I have added another PR with alternative implementation according to @pavel-d's suggestion: https://github.com/travis-ci/travis-migrations/pull/218

**Comment by shairyar** [comment]
Sorry for the late response, I think I agree with @pavel-d suggestion.

**Comment by artursmirnov** [comment]
Thank you @shairyar, then I'm closing this PR in favor of https://github.com/travis-ci/travis-migrations/pull/218. Thanks everyone for reviews and constructive feedback 👍

**Pull request closed without merging**